src/Core/Module/HookRepository.php

```diff
@@ -156,6 +157,20 @@ public function persistHooksConfiguration(array $hooks)
 
                 $this->db->insert('hook_module', $row);
 
+                if (!empty($extra_data['only_pages'])) {
+                    $extra_data['except_pages'] = [];
+
+                    $controllersNames = Dispatcher::getControllersNames(_PS_FRONT_CONTROLLER_DIR_);
```

Comment:
Maybe you should add ModuleFrontController too 🤔
Because some module controllers are important like for a blog module or discounted product listing for example

Reply:
You're right, I will add modules front controllers too.

Reply:
After few research, at this time `except_pages` **seems to don't works** with all module controllers because some of thoses don't have `controller_name` or `php_self` (see `classes/Hook.php` **line 858**) attributes set (eg: module `ps_emailsubscription/controllers/front/verification.php` results in `module-ps_emailsubscription-` key).

**I think an issue need to be open** to fix module controller name to have something even (maybe #19226) before we can include module controller in this PR.

@Matt75 **If you have any suggestions** on how to get all modules controllers names? I tried with filenames return by `Dispatcher::getModuleControllers()` but it doesn't works when checking permissions in `Hook::exec()`.